### PR TITLE
tests/rule: make TestLargeRules more stable

### PR DESCRIPTION
**Pull request by HuSharp**
### What problem does this PR solve?

Issue Number: Ref #7969

### What is changed and how does it work?

tested by #7969
Because below code will add default rule after the previous test finished, we need to skip default rule for large rule test to make it stable
https://github.com/tikv/pd/blob/7e95abff62564dbe683ee2359aa69769cab4dc0d/tests/server/api/rule_test.go#L62-L78 will 

```commit-message
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

**Comment by rleungx** [comment]
Is there any error stack?

**Comment by HuSharp** [comment]
> Is there any error stack?

This condition will never be met, I printed `respBundle` which has one more default value `{GroupID: "pd", ID: "default", Role: "voter", Count: 3}  than resp (this default value is added by `TearDownTest`)
(https://github.com/tikv/pd/blob/7e95abff62564dbe683ee2359aa69769cab4dc0d/tests/server/api/rule_test.go#L1271-L1273)

File: tests/server/api/rule_test.go [hunk 1]
```diff
@@ -1261,6 +1261,13 @@ func (suite *ruleTestSuite) postAndCheckRuleBundle(urlPrefix string, bundle []pl
 		err = tu.CheckGetJSON(testDialClient, urlPrefix+"/config/placement-rule", nil,
 			tu.StatusOK(re), tu.ExtractJSON(re, &respBundle))
 		re.NoError(err)
+		// skip default rule check
```
Comment (rleungx) [inline tests/server/api/rule_test.go:1264]: Will it affect `TestDeleteAndUpdate`?

**Comment by HuSharp** [comment]
> > will add default rule after the previous test finished
> 
> Why does it happen? Does it share the same environment?

yes, this suite test build env firstly and then using `TearDownTest` to clean up placement rules

https://github.com/tikv/pd/blob/7e18a69734f2ae391516625ec0230c05b363f5c5/tests/server/api/rule_test.go#L51-L56

**Comment by HuSharp** [comment]
/hold

meet other issues need to be fixed

**Comment by HuSharp** [comment]
Root cause is 
- ruleTestSuite has run some heavy load tests and reusing the same environment
- we run `suiteTest` just by `1 CPU` resulting in the cluster maybe meeting leader transfer during the whole suite test
- I have allocated suiteTest more resources to avoid unnecessary error https://github.com/tikv/pd/pull/7991/files#diff-d1de7e6511458d33a769c0c501ec0ea45494dd9d2e830dc315e8fe623eb34b76R594-R598

So, we can close this pr now :)

**Pull request closed without merging**